Sources/ApolloWebSocket/DefaultImplementation/WebSocket.swift

```diff
@@ -183,8 +199,12 @@ public final class WebSocket: NSObject, WebSocketClient, StreamDelegate, WebSock
     return canWork
   }
 
-  /// Used for setting protocols.
-  public init(request: URLRequest) {
+  /// Designated initializer.
+  ///
+  /// - Parameters:
+  ///   - request: A URL request object that provides request-specific information such as the URL.
+  ///   - webSocketProtocol: Protocol to use for communication over the web socket.
+  public init(request: URLRequest, webSocketProtocol: WSProtocol) {
```

Comment:
I'm fine with leaving this as is if you want, but I personally would have probably just named these parameters `protocol: WSProtocol`. We are already in the context that this is a parameter to the initialization of a `WebSocket` and the type name is `WSProtocol`, so having the parameter named `webSocketProtocol` seems redundant to me.

But this is a minor nit.

Reply:
I've no objection to renaming them, good point about the enclosing context for a shorter parameter name.